backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/ui/web/order/BOMExploderCommand.java

```diff
@@ -130,17 +139,59 @@ public BOMExploderCommand(@Nullable final BOMUse bomToUse,
 			final OrderLineCandidate lineCandidate = initialCandidate.toBuilder()
 					.productId(bomLineProductId)
 					.attributes(attributes)
-					.qty(bomLineQty)
+					.qty(Quantitys.create(bomLineQty, bomUomId))
 					.compensationGroupId(compensationGroupId)
-					.uomId(bomUomId)
 					.explodedFromBOMLineId(bomLineId)
 					.build();
 			result.addAll(this.toBuilder()
 					.initialCandidate(lineCandidate)
 					.build()
-					.execute());  // recurse
+					.execute0());  // recurse
 		}
 
 		return result;
 	}
+
+	private List<OrderLineCandidate> reduceCandidates(final List<OrderLineCandidate> result)
+	{
+		final ArrayList<OrderLineCandidate> reduced = new ArrayList<>();
+		for (final OrderLineCandidate candidate : result)
+		{
+			if (candidate.getQty().signum() == 0)
+			{
+				continue;
+			}
+			if (candidate.getQty().signum() < 0)
+			{
+				throw new AdempiereException("Invalid qty: " + candidate.getQty());
+			}
+			if (candidate.getQty().toBigDecimal().compareTo(BigDecimal.ZERO) == 0)
```

Comment:
why do we have to check `candidate.getQty().toBigDecimal().compareTo(BigDecimal.ZERO) == 0`
when e already checked `candidate.getQty().signum() == 0` ?

Reply:
If I am 100% honest, it was the Copilot that suggested most of the contents of this function. I thought it was beautiful, even if slightly overkill. Removed.